File: 3010. Divide an Array Into Subarrays With Minimum Cost I.rb
#* 3010. Divide an Array Into Subarrays With Minimum Cost I
nums = [1,2,3,12]
# @param {Integer[]} nums
# @return {Integer}
def minimum_cost(nums)
  nums.first + nums[1, nums.last].min(2).sum
end
p minimum_cost(nums)
# @param {Integer[]} nums
# @return {Integer}
def minimum_cost(nums)
  nums[0] + nums[1..].min(2).sum  
end

def minimum_cost(nums)
  one = nums.shift
  nums.delete(nums.max)
  nums.sum + one
end

File: 3010. Divide an Array Into Subarrays With Minimum Cost I.py
#PYTHON
#* 3010. Divide an Array Into Subarrays With Minimum Cost I
nums = [1,2,3,12]
from typing import List
class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        return nums[0] + sum(sorted(nums[1:])[:2])
    
print(Solution().minimumCost(nums))
        
